backends/backend_libkcapi.c

```diff
@@ -0,0 +1,2365 @@
+                goto out;
+        }
+
+        /* Setting length of associated data */
+        kcapi_aead_setassoclen(handle, assoclen);
+
+        /* Padding IV to a size of 12 bytes */
+        ret = kcapi_pad_iv(handle, data->iv.buf, data->iv.len, &newiv, &newivlen);
+        if (ret)
+        {
+                logger(LOGGER_ERR, "gcm: iv padding failed\n");
+                goto out;
+        }
+
+        ret = -ENOMEM;
+        outbuflen = kcapi_aead_outbuflen_enc(handle, datalen, assoclen, taglen);
+        inbuflen = kcapi_aead_inbuflen_enc(handle, datalen, assoclen, taglen);
+        buf = calloc(1, outbuflen);
+        if (!buf)
+        {
+                logger(LOGGER_ERR, "gcm: allocation of buf failed\n");
+                goto out;
+        }
+
+        kcapi_aead_getdata_output(handle, buf, outbuflen, 1,
+                                        &assoc, &assoclen,
+                                        &o_data, &datalen,
+                                        &tag, &taglen);
+
+        /* Seting key */
+        if (!data->key.len || !data->key.buf || kcapi_aead_setkey(handle, data->key.buf, data->key.len))
+        {
+                logger(LOGGER_ERR, "gcm: key setting failed\n");
+                goto out;
+        }
+
+        if(data->assoc.buf)
+                memcpy(assoc, data->assoc.buf, assoclen);
+
+        if(data->data.buf)
+                memcpy(o_data, data->data.buf, datalen);
+
+        /* Performing encryption */
+        ret = kcapi_aead_encrypt(handle, buf, inbuflen,
+                                        newiv,
+                                        buf, outbuflen,
+                                        KCAPI_ACCESS_SENDMSG);
+        if (ret < 0)
+        {
+                logger(LOGGER_ERR, "gcm: cipher operation of buffer failed: %d %d\n", errno, ret);
+                goto out;
+        }
+        else if ((uint32_t)ret != outbuflen)
+        {
+                logger(LOGGER_ERR, "gcm: received data length %d does not match expected length %u\n", ret, outbuflen);
+                goto out;
+        }
+
+        free_buf(&(data->iv));
+
+        if(data->data.buf)
+                free_buf(&(data->data));
+        if(data->tag.buf)
+                free_buf(&(data->tag));
+
+        CKINT_LOG(alloc_buf(datalen, &data->data), "gcm: ct buffer could not be allocated\n");
+
+        if(o_data)
+                memcpy(data->data.buf, o_data, datalen);
+
+        CKINT_LOG(alloc_buf(taglen, &data->tag), "gcm: tag buffer could not be allocated\n");
+
+        if(tag)
+                memcpy(data->tag.buf, tag, taglen);
+out:
+        kcapi_aead_destroy(handle);
+        if (newiv)
+                free(newiv);
+        if (buf)
+                free(buf);
+        return ret;
+}
+
+static int gcm_decrypt(struct aead_data *data, flags_t parsed_flags)
+{
+        struct kcapi_handle *handle = NULL;
+        char cipher[] = "gcm(aes)";
```

Comment:
As discussed, please use existing function to return the name, instead of hardcoding.